agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/StartupDiagnostics.java

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.PidFinder;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+
+class StartupDiagnostics {
+
+  // For Java >= 9
+  // Execute with -XX:+UnlockDiagnosticVMOptions -XX:CompilerDirectivesFile=path/jvm-disable-c2.json
+  // Content of jvm-disable-c2.json file:
+  // [
+  //  {
+  //    match: [
+  //      "*::*"
+  //    ],
+  //    c2: {
+  //      Exclude: true
+  //    }
+  //  }
+  // ]
+  private static final String APPPLICATIONINSIGHTS_DEBUG_JIT_C_2_DISABLING_ENABLED =
+      "appplicationinsights.debug.jit.c2-disabling.enabled";
```

Comment:
this option name is a bit confusing, what do you think of `applicationinsights.experiment.clear-compiler-directives-after-initialization`

(and differentiating `debug` meaning just logging, from `experiment` meaning an experimental option)

Reply:
Agree, renaming done

Reply:
I have also put the code related to the JVM compiler in a specific class (single responsibility principle).